services/src/main/java/org/keycloak/vault/AbstractVaultProvider.java

```diff
@@ -99,7 +99,7 @@ private void checkForLegacyKey(VaultKeyResolver resolver, String vaultSecretId,
      * @return a boolean indicating whether the validation passed.
      */
     protected boolean validate(VaultKeyResolver resolver, String key, String resolvedKey) {
-        if (key.contains(File.separator)) {
+        if (key.indexOf('/') >= 0 || key.indexOf(File.separatorChar) >= 0) {
```

Comment:
This way on Unix-like systems, `\` will be accepted (though it won't matter as it's not a path separator there). But if someone e.g. moves their KC instance to Windows, then it starts failing which might be confusing.

I'd suggest making it consistent and check for both `/` and `\`, on top of `File.separator` just in case...

Reply:
I addressed the suggestion and added a release note as well.